store/postgres/user.go

```diff
@@ -517,3 +533,439 @@ func (us *UserStore) DeleteAccount(ctx context.Context, uid string) error {
+	}
+	defer setPicture.Close()
+
+	result, err := setPicture.Exec(uid, pict)
+	rowsAff, _ := result.RowsAffected()
+
+	if err != nil || rowsAff != 1 {
+		log.Error().Err(err).Msg("error inserting user session")
+		return errors.New("failed to add user picture")
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		log.Error().Err(err).Msg("error committing changes")
+		return errors.New("failed to add user picture")
+	}
+
+	return nil
+}
+
+func (us *UserStore) SetUserSession(ctx context.Context, t store.TokenDetails, uid string, ip string, device string) error {
+	var tx *sql.Tx
+	tx, err := us.db.Begin()
+
+	if err != nil {
+		log.Error().Err(err).Msg("failed to begin transaction")
+		return errors.New("failed to set user session")
+	}
+
+	defer tx.Rollback()
+	defer func() {
+		if rollBackErr := tx.Rollback(); rollBackErr == nil {
+			log.Error().Err(err).Msg("rolling back changes")
+		}
+	}()
+
+	var setSession *sql.Stmt
+	setSession, err = tx.Prepare(`INSERT into SESSION (id, userid, ip_address, created_at, device) VALUES ($1,$2,$3,$4,$5)`)
+	if err != nil {
+		log.Error().Err(err).Msg("error preparing statement")
+		return errors.New("failed to set user session")
+	}
+	defer setSession.Close()
+
+	result, err := setSession.Exec(t.AccessUuid, uid, ip, time.Now(), device)
+	rowsAff, _ := result.RowsAffected()
+
+	if err != nil || rowsAff != 1 {
+		log.Error().Err(err).Msg("error inserting user session")
+		return errors.New("failed to set user session")
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		log.Error().Err(err).Msg("error committing changes")
+		return errors.New("failed to set user session")
+	}
+
+	return nil
+}
+
+func (us *UserStore) GetUserSession(ctx context.Context, uid string, sessionId string) (store.UserSession, error) {
+	psqlStatement1 := `SELECT ip_address, created_at, updated_at FROM session WHERE id = $1`
+	psqlStatement2 := `SELECT id, device FROM session WHERE userid = $1`
+	var userSessionResponse store.UserSession
+	var userInfo store.UserInfo
+	var ip string
+	var created_at time.Time
+	var updated_at sql.NullTime
+	var tmpUat time.Time
+
+	// first sql query to get the user current session info
+	res1 := us.db.QueryRow(psqlStatement1, sessionId)
+	err := res1.Scan(&ip, &created_at, &updated_at)
+	if err != nil {
+		log.Error().Err(err).Msg(err.Error())
+		return userSessionResponse, errors.New("unable to find the user session")
+	}
+
+	userSessionResponse.Ip = ip
+	userSessionResponse.Created_at = created_at
+	userSessionResponse.Is_current = true
+
+	if !updated_at.Valid {
+		userSessionResponse.Updated_at = tmpUat
+	} else {
+		var tmp time.Time
+		err := us.db.QueryRow(`SELECT updated_at from SESSION where id = $1`, sessionId).Scan(&tmp)
+		if err != nil {
+			log.Error().Err(err).Msg(err.Error())
+			return userSessionResponse, errors.New("unable to find the user session")
+		}
+		userSessionResponse.Updated_at = tmp
+	}
+
+	// Second sql query to get the client info
+	res2, err := us.db.Query(psqlStatement2, uid)
```

Comment:
`db.Query` return `Rows` that must be closed
below this, after checking err != nil , do 
```
defer res2.Close()
```

similar comment for other code that uses `Query` or `QueryContext`